lib/setup.rb

```diff
@@ -1,10 +1,40 @@
 class Setup
-  def initialize(board, ships)
-    @board = board
-    @ships = ships
+  attr_reader :player,
+              :board
+
+  def initialize(board, ships, player)
+    @board  = board
+    @ships  = ships
+    @player = player
+  end
+   # it will be a string concatenation of a sample of A..D and a sample of 1..4
+    # it will need the ships to check ship length to check how many random cells are needed
+    # for each cell, it will randomly generate a letter and a number and then return array 
+    # while valid_placement = false, loop. 
+
+
+  def place_computer_ship(ship, board)
+    placement = random_coordinates(ship, board)
+    while board.valid_placement?(ship, placement) == false
+      placement = random_coordinates(ship, board)
+    end 
+    placement
+  end
+
+  def random_coordinates(ship, board)
```

Comment:
I suppose we could still add a test for this method to check validity, but it kind of gets covered in the other test, since we refactored this out